### PR TITLE
Make use of NoHash hashing for FileId and CrateId

### PR DESCRIPTION
**Pull request by Veykril**
Both of these are mere integers so there is nothing to hash here.

Ideally we would use this for `la_arena::Idx` too, but that doesn't work due to the orphan rule, and `la_arena` is unfortunately a public library so we can't really do much here... Unless we remove the trait restriction but I'd like not to

### CODE REVIEW

File: crates/stdx/src/lib.rs [hunk 1]
```diff
@@ -7,6 +7,7 @@ use std::{cmp::Ordering, ops, time::Instant};
 use std::{io as sio, iter};
 
 mod macros;
+pub mod hash;
```
Comment (lnicola) [inline crates/stdx/src/lib.rs:10]: Is this missing?

Comment (Veykril) [reply crates/stdx/src/lib.rs:10]: it is :) forgot the `-u` flag

**Comment by Veykril** [comment]
More non-deterministic tests 🎉

**Comment by Veykril** [comment]
This could degrade performance if the hashmap doesn't like sequential hashes, but afaik hashbrown should be fine with this.

**Pull request merged**